static/js/unread.js

```diff
@@ -399,6 +399,27 @@ class UnreadTopicCounter {
 
         return id_set.size !== 0;
     }
+
+    topic_has_any_unread_mention_messages(stream_id, topic) {
+        const per_stream_bucketer = this.bucketer.get_bucket(stream_id);
+        if (!per_stream_bucketer) {
+            return false;
+        }
+
+        const topic_bucket = per_stream_bucketer.get_bucket(topic);
+        if (!topic_bucket) {
+            return false;
+        }
+
+        const ids = Array.from(topic_bucket);
+
+        for (const id of ids) {
+            if (unread_mentions_counter.has(id)) {
+                return true;
+            }
+        }
```

Comment:
@timabbott , I'm sure about is there any better approach through which we can get this thing done in less than `O(n)`.
Would like to see your suggestions, if any.

Reply:
This is again an issue of having the wrong API. What you want to do is have a function `get_topics_with_unread_mentions(stream_id)` that returns a `Set` (and will be just as fast as this single-topic version), and call that outside the loop in `get_list_info`, and then inside the loop, you just need to check if the topic is inside the set, which is `O(1)`.

Reply:
Updated the PR removing this approach and tried to add a new one which does this operation in `O(1)`.
Thanks !